File: backend.js

const API_KEY ='61abe2de00cb61a5e27c15dfc824dc67'

function enviar() {
    const cidade = document.getElementById('location-input').value;
    if (cidade) {
        fetch(`https://api.openweathermap.org/data/2.5/weather?q=${cidade}&appid=${API_KEY}&lang=pt_br&units=metric`)
            .then(response => response.json())
            .then(data => {
                if (data.cod === 200) {
                    document.getElementById('location-input').value = `${data.name}, ${data.sys.country}`;
                    document.getElementById('location').textContent = `${data.name}, ${data.sys.country}`;
                    document.getElementById('current-temp').textContent = `${(data.main.temp).toFixed(1)} °C`;
                    document.getElementById('description').textContent = capitalizeFirstLetter(data.weather[0].description);
                } else {
                    console.error('Erro ao carregar os dados:', data.message);
                }
            })
            .catch(error => console.error('Erro ao carregar os dados:', error));
    } else {
        console.error('Por favor, insira o nome de uma cidade.');
    }
}


function capitalizeFirstLetter(string) {
    return string.charAt(0).toUpperCase() + string.slice(1);
}

function formatarDoisDigitos(numero) {
    return numero.toString().padStart(2, '0');
}

function atualizarHora() {
    const agora = new Date();
    document.getElementById('hora').innerHTML = formatarDoisDigitos(agora.getHours());
    document.getElementById('minutos').innerHTML = formatarDoisDigitos(agora.getMinutes());
    document.getElementById('segundos').innerHTML = formatarDoisDigitos(agora.getSeconds());
    document.getElementById('dia').innerHTML = formatarDoisDigitos(agora.getDate()); 
    document.getElementById('mes').innerHTML = formatarDoisDigitos(agora.getMonth() + 1); 
    document.getElementById('ano').innerHTML = formatarDoisDigitos(agora.getFullYear().toString().slice(-2)); 
}

atualizarHora();

setInterval(atualizarHora, 1000);

File: backend.py
import requests

from translate import Translator

API_KEY = "61abe2de00cb61a5e27c15dfc824dc67"

cidade = "Brasília"

link = f"https://api.openweathermap.org/data/2.5/weather?q={cidade}&appid={API_KEY}&lang=pt_br&units=metric"

requisicao = requests.get(link)
print(requisicao.json())
requisicao_dic=requisicao.json()
descricao = requisicao_dic['weather'][0]['description']
temperatura = requisicao_dic['main']['temp']
humidade = requisicao_dic['main']['humidity']


print(f'{cidade} está com {descricao}, a temperatura nesse momento é de {temperatura:.1f} °C, com {humidade}% de humidade relativa do ar.')
link = "https://api.kanye.rest"
requisicao = requests.get(link)
requisicao_dic = requisicao.json()
frase = requisicao_dic['quote']

# Traduzindo a frase para o português do Brasil
tradutor = Translator(to_lang="pt")
frase_traduzida = tradutor.translate(frase)

# Exibindo a frase traduzida
print(f'A frase do dia é: {frase_traduzida}')